### ISSUE TITLE
`check` has no per-task bond, so a wedged task eats the job and names nothing — and the job now runs at its 10-minute limit

### ISSUE THREAD

**Issue by zernie** [zernie/agent-lint#209]
Two related facts, both measured on 2026-09-03.

## 1. A wedged task is reported as a bare `cancelled`

On PR #196 the `check` job died at its 10-minute job bond with `npm ci` still running:

```
✓ format (21.3s)
✓ lint (32.8s)
##[error]The operation was canceled.
…
Terminate orphan process: pid (3123) (npm run check)
Terminate orphan process: pid (3610) (npm ci)
```

A **job-level** `timeout-minutes` is reported exactly like that — `cancelled`, no stated reason. It reads identically to an externally cancelled run, and I first misdiagnosed it as one. A cancelled job is neither green nor red, so it gets investigated by hand every time.

**A workflow step-level bond would not help.** That `npm ci` is not a separate step: it is the `rule-enforcer` task inside `scripts/check.mjs` —

```js
{ name: "rule-enforcer", cmd: "npm ci --prefix rule-enforcer && node rule-enforcer/gate.js" }
```

so a bond on `run: npm run check` bounds exactly the same thing and still names nothing.

## 2. The job is now running at its bond, not near its recorded cost

`ci.yml` carries `timeout-minutes: 10` beside the comment `# measured 2m36s.` The successful re-run took **10:00** (22:44:26 → 22:54:26). Whatever the cause, `check` has grown ~4× past the number its bond was sized against, so it will keep flapping between "slow" and "killed".

## Proposed

A **per-task** bond in `scripts/check.mjs`'s `run()`, so a hang fails with the task's own name instead of charging the job. Prototyped and verified firing (a planted `sleep 600` task with a short bond is killed and reported), then deliberately NOT shipped in #196 — that PR was a docs correction and otherwise green, and pushing an unvalidated CI change into a green PR is how a cycle gets lost.

Shape that worked:

- one uniform ceiling (5 min/task) rather than a per-task exception list — a hand-maintained list of "the slow ones" is the thing `check.mjs` exists to abolish
- resolve with the shell's conventional `124` and a line saying the task was killed rather than failing on its own merits
- honest limit: killing the shell does not reap a grandchild `npm ci`; the runner reaps it at cleanup. The value is the verdict and the name, not process hygiene

Worth doing alongside: find out **why** `check` grew to 10 minutes. The bond change makes the failure legible; it does not make the job fast.
